client/src/components/Gallery.jsx

```diff
@@ -1,29 +1,56 @@
 import React from 'react';
 import Carousel from './Carousel.jsx';
-import {mount, shallow} from 'enzyme';
+
+const testImages = ['https://loremflickr.com/320/240/foods?random=1',
+'https://loremflickr.com/320/240/foods?random=2', 'https://loremflickr.com/320/240/foods?random=3', 'https://loremflickr.com/320/240/foods?random=4', 'https://loremflickr.com/320/240/foods=5', 'https://loremflickr.com/320/240/foods?random=6', 'https://loremflickr.com/320/240/foods?random=7', 'https://loremflickr.com/320/240/foods?random=8', 'https://loremflickr.com/320/240/foods?random=9', 'https://loremflickr.com/320/240/foods?random=10']
 
 class Gallery extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      imageUrl: 'https://loremflickr.com/320/240/foods'
+      photos: testImages,
+      mainPhoto: testImages[0],
+      mainPhotoIndex: 0,
+      carouselBeginIndex: 0, // use later for array of photos
+      carouselEndIndex: 3
     }
     this.carouselLeftArrowClick = this.carouselLeftArrowClick.bind(this);
     this.carouselRightArrowClick = this.carouselRightArrowClick.bind(this);
   }
 
   carouselLeftArrowClick() {
-    console.log('left');
+    // console.log('left');
+    let mainPhotoIndex = this.state.mainPhotoIndex;
+    let firstPhoto = mainPhotoIndex === 0;
+    let index = firstPhoto ? 0 : mainPhotoIndex - 1;
+    this.setState({
+      mainPhotoIndex: index
+    }, () => {
+      this.setState({
+        mainPhoto: testImages[this.state.mainPhotoIndex]
+      })
+    })
   }
 
   carouselRightArrowClick() {
-    console.log('right');
+    // console.log('right');
+    let lastIndex = this.state.photos.length - 1;
+    let mainPhotoIndex = this.state.mainPhotoIndex;
+    let lastPhoto = mainPhotoIndex === lastIndex;
+    let index = lastPhoto ? lastIndex : mainPhotoIndex + 1;
+    this.setState({
+      mainPhotoIndex: index
+    }, () => {
+      this.setState({
+        mainPhoto: testImages[this.state.mainPhotoIndex]
```

Comment:
Also, if you're already on the last photo, I would just do nothing to the state, rather than setting it to the same state as before, which will trigger a re-render. So maybe only setState if lastPhoto = false? Same with left click, if you're already on the first photo, just do nothing

Reply:
This is the way I'm handling the last photo of the index otherwise it keeps incrementing and will return undefined. Unless, I can refactor to handle the last photo condition to just return that's a possibility I think.